conf4j-core/src/main/java/org/conf4j/core/RootConfigurationProvider.java

```diff
@@ -0,0 +1,126 @@
+package org.conf4j.core;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.typesafe.config.Config;
+import org.conf4j.core.ext.ConfigurationExtensions;
+import org.conf4j.core.source.ConfigurationSource;
+import org.conf4j.core.source.reload.ReloadStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+
+public class RootConfigurationProvider<T> implements ConfigurationProvider<T> {
+
+    private static final Logger logger = LoggerFactory.getLogger(RootConfigurationProvider.class);
+    private static final String EMPTY_STRING = "";
+
+    private final ChangeListenersNotifier<T> changeListenersNotifier = new ChangeListenersNotifier<>();
+    private final ConfigurationExtensions configurationExtensions = new ConfigurationExtensions();
+    private final AtomicReference<T> configurationCache = new AtomicReference<>();
+
+    private final ObjectMapper mapper;
+    private final Class<? extends T> configurationClass;
+    private final ConfigurationSource configurationSource;
+    private final List<ReloadStrategy> reloadStrategies;
+    private final String configRootPath;
+
+    RootConfigurationProvider(Class<? extends T> configurationClass,
```

Comment:
The name doesn't make sense yet. It's to provide a configuration, matching the configurationClass type, from configurationSouce, with a bunch of reload strategies, only from a specific rootPath. Why only Root is the name chosen here?